### DISCUSSION TITLE
Itch.io support

### DISCUSSION THREAD

**Discussion by pyrokzg** [ImranR98/Obtainium#2047] [Ideas]
Would it be possible to grab updates for software from itch?

**Comment by DarkCrypt** [comment]
Itch.io doesn't seem to be a suitable source in my opinion.

Here's 2 reasons why:
1. Too many inconsistencies in general.
2. Download buttons within the website are loaded with JavaScript which Obtainium can't scrape.

After experimenting a little it still may be possible to use Obtainium to track these apps as "Track-Only". I did manage to add 3 apps this way but I'm concerned about inconsistencies which could easily break the configs.

Have you tried the open-source app called Mitch from F-Droid?
- [Mitch](https://f-droid.org/packages/ua.gardenapple.itchupdater/) is an unofficial mobile client for itch.io that allows you to install Android games from the store and keep them updated.

**Reply by pyrokzg** [reply]
Thanks for the recommendation, I will give Mitch a try. If you do find a way to consistently get update notifications that would also be useful. 
I suspected the site layout and inconsistencies between project formats could cause ongoing issues.

**Reply by DarkCrypt** [reply]
Your welcome friend. To be fully transparent I'm not the dev or a contributor. Just a user who's quite active within Obtainium's issues and discussions on both repos. I attempt to help in whatever ways I can basing my comments on the knowledge and experience I have with Obtainium. So your request will still need to be considered by the appropriate decision makers.

I may be able to provide a track-only config for a game you're interested in if you'd like to try to test this out? All I'd need is the link to the game from itch.io and I'll try to make it work.

**Reply by 5310** [reply]
While not OP, if you could post a track-only example of [this title](https://dragonrod342.itch.io/digital-tamers-2), I would be much obliged!

**Reply by DarkCrypt** [reply]
Absolutely. Here's a track-only config, including a breakdown of your request.

Digital Tamers 2 config (track-only):
Import App -> [Digital Tamers 2](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22519603403%22%2C%22url%22%3A%22https%3A%2F%2Fdragonrod342.itch.io%2Fdigital-tamers-2%22%2C%22author%22%3A%22dragonrod342.itch.io%22%2C%22name%22%3A%22App%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22devlog%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Atrue%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22partialAPKHash%5C%22%2C%5C%22trackOnly%5C%22%3Atrue%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22DigitalTamers02%20-%20(V%5B0-9%5D%2B%5C%5C%5C%5C.%5B0-9%5D%2B%5C%5C%5C%5C.%5B0-9%5D%2B%7C-%5B0-9%5D%2B)%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%241%5C%22%2C%5C%22versionDetection%5C%22%3Atrue%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Atrue%2C%5C%22appName%5C%22%3A%5C%22Digital%20Tamers%202%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22The%20sequel%20to%20the%20acclaimed%20casual%20fighting%20fan%20game%20%5C%5C%5C%22Digital%20Tamers%3A%20ReBorn%5C%5C%5C%22.%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Afalse%7D%22%2C%22overrideSource%22%3Anull%7D)
- App Source URL: `https://dragonrod342.itch.io/digital-tamers-2`
- Override Source: `None` or `HTML`
- Custom APK Link Filter by Regular Expression: `devlog`
- Apply version string extraction Regex to entire page: `Toggle ON`
- Track-Only: `Toggle ON`
- Version String Extraction RegEx: `DigitalTamers02 - (V[0-9]+\.[0-9]+\.[0-9]+|-[0-9]+)`
- Match Group to Use for Version String Extraction RegEx: `$1`
- App Name: `Digital Tamers 2`
- About: `The sequel to the acclaimed casual fighting fan game "Digital Tamers: ReBorn".`

**Reply by 5310** [reply]
Thank you very much for the step-by-step guide as well!
